cmd/thanos/receive.go

```diff
@@ -489,57 +499,69 @@ func runReceive(
 	}
 
 	if upload {
-		level.Debug(logger).Log("msg", "upload enabled")
-		if err := dbs.Sync(context.Background()); err != nil {
-			level.Warn(logger).Log("msg", "initial upload failed", "err", err)
+		logger := log.With(logger, "component", "uploader")
+		upload := func(ctx context.Context) error {
+			level.Debug(logger).Log("msg", "upload starting")
+			start := time.Now()
+			ctx, cancel := context.WithTimeout(ctx, uploadTimeout)
```

Comment:
Again, as I mentioned. This is tricky as this timeout is not per block but per either thousands or one block, so not sure if this timeout is useful / not harmful :thinking:  Worth to make it per block? (on objstore level)?

So something we always wanted: https://github.com/thanos-io/thanos/issues/1830

Reply:
I've tried to address this issue at https://github.com/thanos-io/thanos/pull/2908